### PR TITLE
Add hubspot-api-client v4.0.6

### PR DESCRIPTION
**Pull request by rxm7706**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by rxm7706** [comment]
Linux Error
```
Processing $SRC_DIR
  Added file://$SRC_DIR to build tracker '/tmp/pip-build-tracker-28wqsksq'
  Running setup.py (path:$SRC_DIR/setup.py) egg_info for package from file://$SRC_DIR
  Created temporary directory: /tmp/pip-pip-egg-info-wgzwfkr9
  Preparing metadata (setup.py): started
  Running command python setup.py egg_info
  Traceback (most recent call last):
    File "<string>", line 2, in <module>
    File "<pip-setuptools-caller>", line 34, in <module>
    File "/home/conda/staged-recipes/build_artifacts/hubspot-api-client_1695475617443/work/setup.py", line 31, in <module>
      with open(DIR_PATH + "/VERSION", "r", encoding="utf-8") as f:
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  FileNotFoundError: [Errno 2] No such file or directory: '/home/conda/staged-recipes/build_artifacts/hubspot-api-client_1695475617443/work/VERSION'
  error: subprocess-exited-with-error
```

**Comment by rxm7706** [comment]
@conda-forge/staged-recipes  ready for review

File: recipes/hubspot-api-client/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/hubspot-api-client-{{ version }}.tar.gz
+  sha256: 621195df4ce231a62bb39739384536441fef0abdede0134b84914b19d3033985
+
+build:
+  noarch: python
+  script:
+    - echo {{ version }} > {{ SRC_DIR }}/VERSION
```
Comment (ocefpaf) [inline recipes/hubspot-api-client/meta.yaml:15]: Can you report this upstream? Most likely their sdist is broken and missing that file.

Comment (rxm7706) [reply recipes/hubspot-api-client/meta.yaml:15]: Hi @ocefpaf 
I started with this version as I needed version <5 for robotframework-windows, It looked like upstream current version has this fixed and will not need this patch once this version is published and upgraded to version 6.0.
I will verify and report if required.

**Pull request merged**